pkg/messaging/direct_messaging.go

```diff
@@ -246,7 +241,13 @@ func (d *directMessaging) invokeWithRetry(
 				}
 				return rResp, fmt.Errorf("failed to invoke target %s after %d retries. Error: %w", app.id, attempt-1, rErr)
 			}
+
 			teardown(false)
+			// Remove the resolved name from the cache
+			if app.cacheKey != "" && d.resolverCache != nil {
+				d.resolverCache.Delete(app.cacheKey)
```

Comment:
Should an address be removed only if the error is for an unavailable endpoint?

Reply:
According to the gRPC docs it can also mean a transient network error

Reply:
I've removed it